packages/web-api-send-notification-runner/src/sender/notification-sender.ts

```diff
@@ -1,21 +1,96 @@
         @inject(NotificationSenderConfig) private readonly notificationSenderConfig: NotificationSenderConfig,
         @inject(Logger) private readonly logger: Logger,
+        @inject(PromiseUtils) private readonly promiseUtils: PromiseUtils,
     ) {}
 
     public async sendNotification(): Promise<void> {
+        let pageScanResult: OnDemandPageScanResult;
         const notificationSenderConfigData = this.notificationSenderConfig.getConfig();
-        this.logger.logInfo(`Id: ${notificationSenderConfigData.id}`);
-        this.logger.logInfo(`Reply URL: ${notificationSenderConfigData.replyUrl}`);
+
+        this.logger.logInfo(`Reading page scan run result ${notificationSenderConfigData.scanId}`);
+        pageScanResult = await this.onDemandPageScanRunResultProvider.readScanRun(notificationSenderConfigData.scanId);
+        this.logger.setCustomProperties({ scanId: notificationSenderConfigData.scanId, batchRequestId: pageScanResult.batchRequestId });
+
+        pageScanResult = await this.startSendingNotification(notificationSenderConfigData, pageScanResult);
+
+        this.logger.logInfo(`Writing page notification status to a storage.`);
+        await this.onDemandPageScanRunResultProvider.updateScanRun(pageScanResult);
+
+        this.logger.trackEvent('SendNotificationTaskCompleted');
+    }
+
+    private async startSendingNotification(
+        notificationSenderConfigData: NotificationSenderMetadata,
+        pageScanResult: OnDemandPageScanResult,
+    ): Promise<OnDemandPageScanResult> {
+        let numberOfTries = 1;
+        let isNotificationSent = false;
+        const errors: NotificationError[] = [];
+
+        this.logger.trackEvent('SendNotificationTaskStarted');
+        while (!isNotificationSent && numberOfTries <= 3) {
+            this.logger.logInfo(`Sending notification, try #${numberOfTries}`);
+            let response;
+            try {
+                response = await this.notificationSenderWebAPIClient.postNotificationUrl(notificationSenderConfigData);
+                if (response.statusCode === 200) {
+                    this.logger.trackEvent('SendNotificationTaskSucceeded');
+                    this.logger.logInfo(`Notification sent Successfully!`);
+                    isNotificationSent = true;
+                } else {
+                    this.logger.trackEvent('SendNotificationTaskFailed');
+                    this.logger.logInfo(`Notification sent failed!, statusCode: ${response.statusCode}, body: ${response.body}`);
+                    isNotificationSent = false;
+                    // tslint:disable-next-line: no-unsafe-any
+                    errors.push({ errorType: 'HttpErrorCode', message: response.body });
+                }
+            } catch (e) {
+                this.logger.trackEvent('SendNotificationTaskFailed');
+                this.logger.logError(`Notification sent failed!, error message: ${(e as Error).message}`);
+                isNotificationSent = false;
```

Comment:
you dont have to set isNotification here. this variable is set to true only after a sucessful send anyway.